node/core/av-store/src/lib.rs

```diff
@@ -0,0 +1,555 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Implements a `AvailabilityStoreSubsystem`.
+//!
+
+#![recursion_limit="256"]
+#![warn(missing_docs)]
+
+use std::collections::HashMap;
+use std::io;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use codec::{Encode, Decode};
+use futures::{select, channel::oneshot, FutureExt};
+use kvdb_rocksdb::{Database, DatabaseConfig};
+use kvdb::{KeyValueDB, DBTransaction};
+
+use polkadot_primitives::v1::{
+	Hash, AvailableData, ErasureChunk, ValidatorIndex,
+};
+use polkadot_subsystem::{
+	FromOverseer, SubsystemError, Subsystem, SubsystemContext, SpawnedSubsystem,
+};
+use polkadot_subsystem::messages::AvailabilityStoreMessage;
+
+const LOG_TARGET: &str = "availability";
+
+mod columns {
+	pub const DATA: u32 = 0;
+	pub const NUM_COLUMNS: u32 = 2;
+}
+
+#[derive(Debug, derive_more::From)]
+enum Error {
+	#[from]
+	Erasure(erasure::Error),
+	#[from]
+	Io(io::Error),
+	#[from]
+	Oneshot(oneshot::Canceled),
+	#[from]
+	Subsystem(SubsystemError),
+}
+
+/// An implementation of the Availability Store subsystem.
+pub struct AvailabilityStoreSubsystem {
+	inner: Arc<dyn KeyValueDB>,
+}
+
+fn available_data_key(candidate_hash: &Hash) -> Vec<u8> {
+	(candidate_hash, 0i8).encode()
+}
+
+fn erasure_chunk_key(candidate_hash: &Hash, index: u32) -> Vec<u8> {
+	(candidate_hash, index, 0i8).encode()
+}
+
+#[derive(Encode, Decode)]
+struct StoredAvailableData {
+	data: AvailableData,
+	n_validators: u32,
+}
+
+/// Configuration for the availability store.
+pub struct Config {
+	/// Cache size in bytes. If `None` default is used.
```

Comment:
```suggestion
	/// Total cache size in megabytes. If `None` the default (128 MiB per column) is used.
```
https://github.com/paritytech/parity-common/blob/bbdd0899c934176f1083469e6618f5bb4bb5fc91/kvdb-rocksdb/src/lib.rs#L47-L48